packages/loader/container-loader/src/deltaManager.ts

```diff
@@ -157,6 +170,56 @@ class NoDeltaStream
     public close(): void { this.dispose(); }
 }
 
+/**
+ * This class exists to help diagnose issues described in https://github.com/microsoft/FluidFramework/issues/7312
+ *
+ * This adapter class tracks resume & pause calls to provide ability to debug cases where
+ * someone is holding queue in paused state too long.
+ * The only thing that it does on top of DeltaQueue is tracking all users who called pause()
+ * and making available callstacks for those calls.
+ *
+ * Long term, we better off switching to tracking callers by
+ *  - pause() returning token that is required to be passed into resume()
+ *  - tracking tokens and stacks with tokens
+ * This would be improvement over current system, as today we can't pair pause & resume calls.
+ */
+class DeltaQueueProxy<T> extends EventForwarder<IDeltaQueueEvents<T>> implements IDeltaQueue<T>, IDeltaQueueWriter<T> {
```

Comment:
There is a class called DeltaQueueProxy in `deltaManagerProxy.ts`.  Might want to rename this.

Also, I don't know why  DeltaQueueProxy in `deltaManagerProxy.ts` needs to be exported?